src/common/Formatter.h

```diff
@@ -226,6 +227,7 @@ namespace ceph {
     void dump_format_va(const char *name, const char *ns, bool quoted, const char *fmt, va_list ap) override;
     int get_len() const override;
     void write_raw_data(const char *data) override;
+    void write_bin_data(char * buff,int len) override;
```

Comment:
what is the difference between binary and raw data?

Reply:
the raw_data wont work on binary data (it works on null terminated string)

Reply:
ok, got it.
so, the buff is the input? please make it `const char*`, also change the name to `data` so it is more clear what is the difference (raw_data is not a great name...)